### ISSUE TITLE
Filter example

### ISSUE THREAD

**Issue by chkoar** [chrispy645/weka-pyscript#8]
Is it possible to provide me with an example of a filter that:
1. Filters only the training instances
2. Adds new nominal column with specified values

Regards,
C

**Comment by christopher-beckham** [comment]
Hi,

In regards to #1, do you mean a filter that only processes training instances, and not test instances?Filters in WekaPyScript work just like how they do with WEKA -- you first "train" the filter by passing it training data, and use the trained filter to process training instances and test instances. Obviously, some filters need not be "trained", e.g. a filter that adds random Gaussian noise to each attribute, which is here:

https://github.com/christopher-beckham/weka-pyscript/blob/master/scripts/add-gauss-noise.py

(That also explains why the train() method returns nothing)

I have just committed an example where you add a new attribute which is nominal and takes 3 values a,b,c:

https://github.com/christopher-beckham/weka-pyscript/blob/master/scripts/add-nominal-col.py

For each instance, we sample u ~ uniform(0,2), where u=0 corresponds to the value "a", u=1 corresponds to "b", and u=2 corresponds to "c".

**Comment by chkoar** [comment]
Hi,

> In regards to #1, do you mean a filter that only processes training instances, and not test instances?

Exactly. I want to filter out training instances only.

> I have just committed an example where you add a new attribute which is nominal and takes 3 values a,b,c: https://github.com/christopher-beckham/weka-pyscript/blob/master/scripts/add-nominal-col.py

Thank you very much. That was what I was looking for.

Regards, 
C

**Comment by christopher-beckham** [comment]
You would have to filter your training set in the Preprocessing tab beforehand, rather than use a FilteredClassifier, which would apply the filter to both the training and test set.

**Comment by chkoar** [comment]
> You would have to filter your training set in the Preprocessing tab beforehand, rather than use a FilteredClassifier, which would apply the filter to both the training and test set.

Ok, thanks. I asked because I liked the idea to use the PyScriptFilter in the experimenter.

**Comment by christopher-beckham** [comment]
Ah, I see. So you would be running something in the Experimenter using cross-validation? If you want, I could add a feature in so that it doesn't run the filter on the test set.

**Comment by chkoar** [comment]
Exactly. It would be a very nice feature. I could quick experiment with reduction and selection methods in Python.

**Comment by christopher-beckham** [comment]
It might be more appropriate to actually make a version of the FilteredClassifier implementation so that it doesn't apply the filter to instances at test time. This should be a trivial thing to do. I can write this if you want.
